presto-parquet/src/main/java/io/prestosql/parquet/writer/MessageTypeConverter.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.parquet.writer;
+
+import org.apache.parquet.format.ConvertedType;
+import org.apache.parquet.format.FieldRepetitionType;
+import org.apache.parquet.format.SchemaElement;
+import org.apache.parquet.format.Type;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.TypeVisitor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class MessageTypeConverter
+{
+    private MessageTypeConverter() {}
+
+    static List<SchemaElement> toParquetSchema(MessageType schema)
+    {
+        List<SchemaElement> result = new ArrayList<>();
+        addToList(result, schema);
+        return result;
+    }
+
+    private static void addToList(final List<SchemaElement> result, org.apache.parquet.schema.Type field)
```

Comment:
I don't think the `final` is required here anymore.  We don't put final on parameters (or local variables) unless necessary.